### PR TITLE
Marionette: migrate to TS

### PR DESCRIPTION
**Pull request by gaetanmaisse**
Issue: Part of #5030

## What I did

Fastly migrate `@storybook/marionette` to TS as I want to close the big TS Migration Issue before moving to improve types of `@storybook/core`.

## How to test

- CI should be 🟢 , we rely only on examples as there is no E2E for Marionette.

### CODE REVIEW

File: app/marionette/src/client/preview/element_check.ts [hunk 1]
```diff
@@ -8,10 +8,10 @@ const allMarionetteViewConstructors = [
 ];
 const viewConstructorsSupportedByMarionette = allMarionetteViewConstructors
   .filter((constructorName) => constructorName in Marionette)
-  .map((constructorName) => Marionette[constructorName]);
+  .map((constructorName) => (Marionette as any)[constructorName]);
```
Comment (gaetanmaisse) [inline app/marionette/src/client/preview/element_check.ts:11]: I tried to find the correct type but it's too 🤯  for a non-marionette user.

File: app/marionette/src/client/preview/index.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ export const {
 } = clientApi;
 
 const framework = 'marionette';
-export const storiesOf = (...args) => clientApi.storiesOf(...args).addParameters({ framework });
-export const load = (...args) => coreLoad(...args, framework);
+export const storiesOf = (...args: any) =>
+  clientApi.storiesOf(...args).addParameters({ framework });
+export const load = (...args: any) => coreLoad(...args, framework);
```
Comment (gaetanmaisse) [inline app/marionette/src/client/preview/index.ts:20]: Will be properly type as soon as I will rework some `@storybook/core` types

File: app/marionette/src/server/framework-preset-marionette.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Configuration } from 'webpack';
+
+export function webpack(config: Configuration) {
+  return config;
+}
```
Comment (gaetanmaisse) [inline app/marionette/src/server/framework-preset-marionette.ts:6]: Not sure we need this but I didn't want to make changes to production code other than a simple TS migration.

**Comment by gaetanmaisse** [comment]
I think we are good to go with this one @ndelangen @shilman, not sure the diff in chromatic are relevant/related to this PR.

**Review by ThibaudAV — APPROVED** [review]
LGTM

IMHO take time to add robust typescript is interesting if users (+framework) are also adepts. if they are users of "js" this is not (or less) necessary 🙊

**Pull request merged**